src/main/java/by/exadel/internship/exception_handing/GlobalExceptionHandler.java

```diff
@@ -36,10 +35,23 @@
     public ResponseEntity<IncorrectData> handleException(
             Exception exception) {
         UUID uuidErrorCode = UUID.randomUUID();
+        System.out.println(exception.getClass());
         log.error("Message: " + exception.getMessage() + " Error UUID code: " + uuidErrorCode);
         IncorrectData data = new IncorrectData();
         data.setMessage(exception.getMessage());
         data.setErrorCode(uuidErrorCode);
         return new ResponseEntity<>(data, HttpStatus.BAD_REQUEST);
     }
+
+    @ExceptionHandler
+    public ResponseEntity<IncorrectData> handleException(
+            BadCredentialsException exception){
+        UUID uuidErrorCode = UUID.randomUUID();
```

Comment:
You can move repeatable code with generation of IncorrectData object to separate common method with return type IncorrectData and reuse it in both methods.

Reply:
Dine